packages/rum/test/userActionCollection.spec.ts

```diff
@@ -0,0 +1,358 @@
+import { Observable, RequestDetails } from '@datadog/browser-core'
+import { LifeCycle, LifeCycleEventType } from '../src/lifeCycle'
+import { UserActionType } from '../src/rum'
+import { $$tests, ChangeEvent, getUserActionId, startUserActionCollection } from '../src/userActionCollection'
+const { newUserAction, trackPageChanges, resetUserAction } = $$tests
+
+function mockClock() {
+  beforeEach(() => {
+    jasmine.clock().install()
+    jasmine.clock().mockDate()
+    spyOn(performance, 'now').and.callFake(() => Date.now())
+  })
+
+  afterEach(() => {
+    jasmine.clock().uninstall()
+  })
+
+  return {
+    tick(ms: number) {
+      jasmine.clock().tick(ms)
+    },
+    expire() {
+      jasmine.clock().tick(100_000)
+    },
+  }
+}
+
+function eventsCollector<T>() {
+  const events: T[] = []
+  beforeEach(() => {
+    events.length = 0
+  })
+  return {
+    events,
+    pushEvent(event: T) {
+      events.push(event)
+    },
+  }
+}
```

Comment:
What about replacing `event` by `userAction`?

Reply:
That's not necessarily user actions, I use it for page activity events too.